pkg/tests/ossm/deploy_on_infra_test.go

```diff
@@ -0,0 +1,183 @@
+spec:
+  config:
+    nodeSelector:
+      node-role.kubernetes.io/infra: ""
+    tolerations:
+    - effect: NoSchedule
+      key: node-role.kubernetes.io/infra
+      value: reserved
+    - effect: NoExecute
+      key: node-role.kubernetes.io/infra
+      value: reserved
+`)
+
+			t.LogStep(fmt.Sprintf("Verify operator pod is running on the infra node. Node expected: %s", workername))
+			retry.UntilSuccess(t, func(t test.TestHelper) {
+				locator := pod.MatchingSelector("name=istio-operator", "openshift-operators")
+				operatorPod := locator(t, oc.DefaultOC)
+				oc.WaitPodReady(t, locator)
+				shell.Execute(t,
+					fmt.Sprintf(`oc get pod -n openshift-operators %s -o jsonpath='{.spec.nodeName}'`, operatorPod.Name),
+					assert.OutputContains(
+						workername,
+						"Success: Operator pod is running on the infra node",
+						"Error: Operator pod is not running on the infra node"))
+			})
+		})
+
+		t.NewSubTest("control plane").Run(func(t TestHelper) {
+			t.Log("Testing: Run OSSM Operator on infra nodes")
+			t.Cleanup(func() {
+				oc.RecreateNamespace(t, meshNamespace)
+				oc.ApplyString(t, meshNamespace, util.RunTemplate(GetSMCPTemplate(env.GetDefaultSMCPVersion()), Smcp))
+				oc.WaitSMCPReady(t, meshNamespace, smcpName)
+			})
+
+			t.LogStep("Patch SMCP to run on infra nodes and wait for the SMCP to be ready")
+			retry.UntilSuccess(t, func(t test.TestHelper) {
+				oc.Patch(t, meshNamespace, "smcp", smcpName, "merge", `
+spec:
+  runtime:
+    defaults:
+      pod:
+        nodeSelector:
+          node-role.kubernetes.io/infra: ""
+        tolerations:
+        - effect: NoSchedule
+          key: node-role.kubernetes.io/infra
+          value: reserved
+        - effect: NoExecute
+          key: node-role.kubernetes.io/infra
+          value: reserved
+`)
+				oc.WaitSMCPReady(t, meshNamespace, smcpName)
+			})
+
+			t.LogStep("Verify that the smcp pods are running on the infra node. Pod expected to be moved: istiod, istio-ingressgateway, istio-egressgateway, jaeger, grafana, prometheus")
+			verifyPodRunningInNode(t)
+		})
+	})
+}
+
+func verifyPodRunningInNode(t TestHelper) {
+	istioPods := []string{"istiod", "istio-ingressgateway", "istio-egressgateway", "jaeger", "grafana", "prometheus"}
+	for _, p := range istioPods {
+		podLocator := pod.MatchingSelector("app="+p, meshNamespace)
+		po := podLocator(t, oc.DefaultOC)
+		retry.UntilSuccess(t, func(t test.TestHelper) {
+			nsPods := getAllPodsFromNode(t, meshNamespace)
+			if !slices.Contains(nsPods, po.Name) {
+				t.Log("The pod is not running on the infra node, delete it and wait for it to be recreated")
+				oc.DeletePod(t, pod.MatchingSelector("app="+p, meshNamespace))
+				nsPods = getAllPodsFromNode(t, meshNamespace)
+				podLocator = pod.MatchingSelector("app="+p, meshNamespace)
+				po = podLocator(t, oc.DefaultOC)
+				if !slices.Contains(nsPods, po.Name) {
+					t.Fatalf("Pod %s is not running on the infra node", p)
+				}
+			}
```

Comment:
I don't understandthis code. What does getAllPodsFromNode do and why do we need it? This code should just check the node of each pod and compare it to the name of the node you tainted in the test.

Reply:
Also, this code should not delete any pods. When the tests patches the SMCP, the operator must move the pods to the infra node without the user having to delete the pods.

Reply:
Ok ok, I can not remember why we need to delete the pod but I'll check the test to directly check the node from each pod